docs/UserGuide.md

```diff
@@ -74,14 +74,71 @@ StonksBook is a **desktop app for managing contacts, optimized for use via a Com
 
 #### Adding a tag: `tag add`
 
+Adds a new customised tag of the specified name. If there is an existing tag with this name, this command will not result in any change in state.
+
+Format: `tag add t/TAG`
+
+* Adds a tag with the specified `TAG` as name. If this tag name already exists in the tag list, there will be no change in the program state. 
+* The `TAG` field must be provided.
+
+Examples:
+
+* `tag add t/friends` adds the tag `friends` to the tag list in StonksBook.
+
 #### Listing all tags: `tag list`
 
+Displays a list of all tags created so far.
+
+![result for 'tag list'](images/tagListMockup.png)
+
+Format: `tag list`
+
+Examples:
+
+* `tag list` displays all tags available in StonksBook.
+
 #### Editing a tag: `tag edit`
 
+Edits an existing tag in the StonkBook to the specified tag name. All entries previously associated with this tag will be updated to associated with the updated tag.
+
+Format: `tag edit INDEX n/NAME`
+
+* Edits the name of the tag at the specified `INDEX` to be the specified `NAME`. The `INDEX` refers to the index number shown in the list displayed by the `tag list` command.
+* The `INDEX` must be a positive integer 1, 2, 3, ...
+* The `NAME` and `INDEX` fields must be provided.
+* All contacts that have been previously associated with this tag will be updated automatically to associate with the updated tag.
+
+Examples:
+
+* `tag edit 1 n/close friends` updates the tag `friends` to `close friends`.
```

Comment:
Should we mention `friends` here? I think it may be better if you just mention that it updates the name of the first tag (as per `tag list`)